include/openssl/core_dispatch.h

```diff
@@ -684,6 +698,10 @@ OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, keymgmt_import_types_ex,
 OSSL_CORE_MAKE_FUNC(const OSSL_PARAM *, keymgmt_export_types_ex,
                     (void *provctx, int selection))
 
+# define OSSL_FUNC_KEYMGMT_RESERVE                    47
+OSSL_CORE_MAKE_FUNC(void *, keymgmt_reserve,
```

Comment:
I would still prefer keymgmt_split instead of reserve.

Reply:
keymgmt_make_holey?

More seriously, keymgmt_reserve_space works for me.  The operation isn't really splitting anything it's poking holes in various places.

Reply:
extract_shard was used by Bouncy castle

Reply:
In a world where we already spounge and squeeze, sprouting would almost be appropriate, wouldn't it? 😆

Reply:
On a serious note, though, I still want to be careful not to make up a name that's too closely tied to HSS.  "reserve" is actually a relatively good name, all things considered (said in spite of my initial scepticism)